### PR TITLE
Implement input action

### LINKED ISSUE #32: Feature Request: Allow entered text as parameter for a callback.
This may be a bit silly since you can prompt a popup or something for it, but maybe have a prompt that can accept text as an argument for a callback. 

Something akin to this: 
```
/// Maybe return a new Command Palette Action for a chain of prompts?
CommandPaletteAction? someFun(String args) { ... }

CommandPaletteAction.nested(
  label: 'Manage Users', childrenActions: [
    CommandPaletteAction.WithCallback(
      label: 'Create', callback: someFun)
  ]
)
```

### PR DESCRIPTION
**Pull request by NishiyamaPedro**

### CODE REVIEW

**Comment by TNorbury** [comment]
Thank you for this pull request, and I apologize that I'm just now seeing it. I will try to take a look at it in the coming weeks.

Would you be able to add one of these types of actions to the example (see [here](https://github.com/TNorbury/command_palette/blob/main/example/lib/main.dart#L67)). That way I can have an idea of how it's supposed to be used.

Thanks!

File: lib/src/controller/command_palette_controller.dart [hunk 1]
```diff
@@ -85,6 +85,9 @@ class CommandPaletteController extends ChangeNotifier {
 
   /// Listens to [textEditingController] and is called whenever it changes.
   void _onTextControllerChange() {
+    if (currentlySelectedAction?.actionType == CommandPaletteActionType.input)
+      return;
```
Comment (TNorbury) [inline lib/src/controller/command_palette_controller.dart:89]: Wrap in brackets
```suggestion
    if (currentlySelectedAction?.actionType == CommandPaletteActionType.input) {
      return;
    }
```

File: lib/src/models/command_palette_action.dart [hunk 2]
```diff
@@ -8,7 +8,9 @@ enum CommandPaletteActionType {
 
   /// Upon being selected a nested action will change the state of the command
   /// palette so that it only shows its children
-  nested
+  nested,
+
+  input
```
Comment (TNorbury) [inline lib/src/models/command_palette_action.dart:13]: Add this comment
```suggestion
  /// Upon being selected the user will be given a text field that they can enter text into and submit.
  /// The entered text will then be passed to the relevant callback
  input
```

**Review by TNorbury — CHANGES_REQUESTED** [review]
Thanks again for this PR. It looks really great, just a few housekeeping items, in addition to what I mentioned above.

**Comment by NishiyamaPedro** [comment]
Thanks for the review. I've made the requested changes and added a basic example.

**Comment by TNorbury** [comment]
Awesome, thanks for the changes!

**Pull request merged**